text/0256-react-cdk-add-jsx-tsx-support.md

```diff
@@ -0,0 +1,182 @@
+
+## Hello ReactCDK
+
+Example ReactCDK application
+
+```jsx
+const { ReactCDK, App, Stack } = require('@aws-cdk/react')
+const Bucket = require('@aws-cdk/react/s3')
+const { CloudFrontDistribution, Origins, DefaultOrigin } = require('@aws-cdk/react/cloudfront')
+const { Api, Resource, Integration } = require('@aws-cdk/react/apigateway')
+const Lambda = require('@aws-cdk/react/lambda')
+
+const EchoLambda = (
+  <Lambda>
+  {
+    async ({event, context}) => ({
+      "statusCode": 200,
+      "headers": {
+            "Content-Type": "application/json"
+        }
+      "body": JSON.stringify(event)
+    })
+  }
+  </Lambda>
+)
+
+const EchoApi = (
+  <Api>
+    <Resource path="*">
+      <Integration type="lambda">
+        <EchoLambda />
+      </Integration>
+    </Resource>
+  </Api>
+)
+
+const WebsiteBucket = () => (
+  <Bucket src="./public" />
+)
+
+const MyCloudFrontDistribution = () => (
+  <CloudFrontDistribution>
+    <Origins>
+      <DefaultOrigin oai="true">
+        <WebsiteBucket />
+      </DefaultOrigin>
+      <Origin path="/api/*">
+        <EchoApi />
+      </Origin>
+    </Origins>
+  </CloudFrontDistribution>
+)
+
+const MyApp = (
+<App>
+  <Stack>
+    <MyCloudFrontDistribution />
+  </Stack>
+</App>
+)
+
+ReactCDK.render(MyApp, {region: 'us-east-1'})
+```
+
+# Motivation
+
+**Why are we doing this?**
+
+Developer accessibility.  Expanding and making the AWS platform accessible
+to a larger group of people.
+
+* Library / framework support is a natural evolution on top of already
+  supported CDK languages.
+* React is familiar to a large group of frontend developers. It's a generalized
+  tech that allows developers to naturally express composition, which is the core
+  tenet of serverless / serviceful solutions.
+* This eases developers into more advanced AWS usage and therefore enabling them
+  to get more value from the platform.
+* Library/framework support on top of already supported language (ts) via jsii
+
+**What use cases does it support?**
+
+The large pool of existing React developers who have experience with React and it's
+component based composition idioms.  Allows them to reuse and leverage their existing
+knowledge to build full stack cloud solutions.
+
+**What is the expected outcome?**
+
+More people leveraging CDK suite -> more innovation and solutions -> advance humanity.
+
+# Design Summary
+
+Code generate a set of ReactCDK components that has feature parity with all TS
+components from cfn constructs to solutions constructs.
+
+# Detailed Design
+
+## Codegen ReactCDK components
+
+A set of ReactCDK components that has feature parity with all TS components from cfn constructs to solutions constructs.
+
+* mapping from react concepts of props, state, children, render props to typescript
+CDK constructors, params, hierarchical nesting (parent), etc.
+* code generated based on the jsii manifests of CDK modules
+[via]((https://twitter.com/emeshbi/status/1305017904027643906?s=20)) @eladb
```

Comment:
This is possible but probably the hardest part. Constructs often offer imperative APIs that won't be expressive JSX but can technically still be accessible from code.

The work we did on [decdk](https://www.npmjs.com/package/decdk) can be leveraged.